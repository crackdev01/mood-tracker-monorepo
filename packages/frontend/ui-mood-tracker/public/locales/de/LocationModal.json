{
  "header": "Ort festlegen",
  "description": "Bitte geben Sie in Ihrem Browser die Erlaubnis, auf Ihren Standort zuzugreifen",
  "buttons": {
    "yes": "Ja",
    "no": "Nein"
  }
}
