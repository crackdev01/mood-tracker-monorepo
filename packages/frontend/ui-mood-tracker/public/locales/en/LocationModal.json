{
  "header": "Set Location",
  "description": "Please provide permission in your browser to access your location"
}
