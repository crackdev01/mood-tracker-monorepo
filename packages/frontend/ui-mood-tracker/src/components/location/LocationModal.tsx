import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Button, Header, Icon, Modal } from 'semantic-ui-react';
import { useTranslation } from 'react-i18next';

import { UserActions } from '../../store/user/types';

const LocationModal = (props: any) => {
  const dispatch = useDispatch();
  const { displayModal, closeModal } = props;
  const { t } = useTranslation(['LocationModal']);
  const [coordinates, setCoordinates] = useState({ lat: 0, long: 0 });

  const setLocationSuccess = (position: any) => {
    setCoordinates({
      lat: position.coords.latitude,
      long: position.coords.longitude,
    });
    dispatch({
      type: UserActions.SET_LOCATION_SUCCESS,
      payload: coordinates,
    });
  };

  const setLocationError = () => {
    dispatch({
      type: UserActions.SET_LOCATION_ERROR,
    });
    closeModal();
  };

  const setLocationDenied = () => {
    dispatch({
      type: UserActions.SET_LOCATION_DENIED,
    });
    closeModal();
  };

  const loadLocation = () => {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(setLocationSuccess, setLocationError, {
        timeout: 30000,
        maximumAge: 75000,
      });
    }
    closeModal();
  };

  return (
    <Modal closeIcon open={displayModal} onClose={closeModal}>
      <Header icon="archive" content={t('header')} />
      <Modal.Actions>
        <Button color="red" onClick={setLocationDenied}>
          <Icon name="remove" /> No
        </Button>
        <Button color="green" onClick={loadLocation}>
          <Icon name="checkmark" /> Yes
        </Button>
      </Modal.Actions>
    </Modal>
  );
};

export default LocationModal;
