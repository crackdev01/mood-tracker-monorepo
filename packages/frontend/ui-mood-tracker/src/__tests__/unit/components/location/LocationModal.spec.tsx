import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LocationModal from '../../../../components/location/LocationModal';

Enzyme.configure({ adapter: new Adapter() });

describe('LocationModal', () => {
  test('renders', () => {
    const w = shallow(<LocationModal />);
    expect(w.debug()).toMatchSnapshot();
  });
});
