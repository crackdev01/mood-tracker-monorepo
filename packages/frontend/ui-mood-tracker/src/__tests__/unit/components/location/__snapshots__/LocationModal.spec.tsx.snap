// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationModal renders 1`] = `
"<Modal closeIcon={true} open={false} onClose={[Function: onClose]} onOpen={[Function: onOpen]} centered={true} dimmer={true} closeOnDimmerClick={true} closeOnDocumentClick={false} eventPool=\\"Modal\\">
  <Header icon=\\"archive\\" content=\\"header\\" />
  <ModalActions>
    <Button color=\\"red\\" onClick={[Function: onClick]} as=\\"button\\">
      <Icon name=\\"remove\\" as=\\"i\\" />
       No
    </Button>
    <Button color=\\"green\\" onClick={[Function: loadLocation]} as=\\"button\\">
      <Icon name=\\"checkmark\\" as=\\"i\\" />
       Yes
    </Button>
  </ModalActions>
</Modal>"
`;
